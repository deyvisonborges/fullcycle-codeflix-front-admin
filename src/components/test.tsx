import * as S from './test.styles'
import { toastActions, useAppDispatch, useAppSelector } from '@/store'

export function Test() {
  const dispatch = useAppDispatch()
  const { isVisible, message } = useAppSelector((state) => state.toast)
  return (
    <div>
      <p>Habilitado: {isVisible}</p>
      {isVisible && <p>Message: {message}</p>}
      <button
        onClick={() =>
          isVisible
            ? dispatch(toastActions.hideToast())
            : dispatch(toastActions.showToast(`Messagem para o toast`))
        }
      >
        {isVisible ? 'Desabilitar' : 'Habilitar'}
      </button>
      <S.Button as="a" href="/dashboard">
        Button
      </S.Button>
      <a href="/not-fount">Not found</a>
    </div>
  )
}
