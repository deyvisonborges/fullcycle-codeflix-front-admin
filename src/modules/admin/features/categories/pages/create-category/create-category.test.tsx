import { CreateCategoryPage } from '.'
import { BrowserRouter } from 'react-router-dom'
import { renderWithProviders } from '@/utils/test/renderWithProviders'
import { mockMatchMedia } from '@/utils/test/mock-matchmedia'

beforeAll(() => mockMatchMedia)

describe('CreateCategoryPage', () => {
  it('renders correctly', () => {
    const { asFragment } = renderWithProviders(<CreateCategoryPage />, {
      wrapper: BrowserRouter
    })
    expect(asFragment()).toMatchSnapshot()
  })
})
