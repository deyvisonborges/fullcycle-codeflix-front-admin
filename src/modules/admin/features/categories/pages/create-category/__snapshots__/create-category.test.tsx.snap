// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCategoryPage renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-ixGGxD oiJVB"
  >
    <div
      class="sc-dntaoT clyXwj"
      id="form__header"
    >
      <div
        class="sc-ghWlax jzNQHp"
      >
        <i
          class="sc-ivxoEo eHFMCT"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M9.707 16.707a1 1 0 01-1.414 0l-6-6a1 1 0 010-1.414l6-6a1 1 0 011.414 1.414L5.414 9H17a1 1 0 110 2H5.414l4.293 4.293a1 1 0 010 1.414z"
              fill-rule="evenodd"
            />
          </svg>
        </i>
        <h2
          class="sc-kLhKbu daeWQb"
        >
          Criar uma nova categoria
        </h2>
      </div>
    </div>
    <form
      class="sc-khLCKb hQCKRV"
    >
      <div
        class="sc-blHHSb blRQuv"
      >
        <div
          class="sc-gtLWhw kFIqRo"
        >
          <label
            class="sc-fAUdSK kTGwm"
            for="name"
          >
            Nome
          </label>
          <input
            class="sc-egkSDF emYusT"
            id="name"
            name="name"
            placeholder="Nome da categoria"
            type="text"
            value=""
          />
        </div>
        <div
          class="sc-gtLWhw kFIqRo"
        >
          <label
            class="sc-fAUdSK kTGwm"
            for="description"
          >
            Descrição
          </label>
          <input
            class="sc-egkSDF emYusT"
            id="description"
            name="description"
            placeholder="Descrição"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="sc-gtLWhw kFIqRo"
      >
        <label
          class="sc-fAUdSK kTGwm"
          for="isActive"
        >
          Ativo?
        </label>
        <input
          class="sc-egkSDF emYusT"
          id="isActive"
          name="isActive"
          type="checkbox"
        />
      </div>
      <a
        href="/dashboard/categories"
      >
        Voltar
      </a>
      <div
        class="sc-dstKZu iccvyj"
      >
        <button
          class="sc-keTIit eEoIei"
          type="button"
        >
          Cancelar
        </button>
        <button
          class="sc-dpBQxM bLsGbw"
          type="submit"
        >
          Salvar
        </button>
      </div>
    </form>
    <footer
      class="sc-jwIPbr bLrzYG"
    />
  </div>
</DocumentFragment>
`;
